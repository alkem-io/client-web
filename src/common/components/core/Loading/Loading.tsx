import { CircularProgress } from '@mui/material';
import React, { FC } from 'react';
import WrapperTypography from '../WrapperTypography';
import { useLoadingStyles } from './Loading.styles';

interface LoadingProps {
  text?: string;
}

export const Loading: FC<LoadingProps> = ({ text = 'Loading' }) => {
  const styles = useLoadingStyles();

  return (
    <div className={styles.container}>
      <CircularProgress className={styles.spinner} />
      <WrapperTypography variant="caption" color="primary" className={styles.text}>
        {text}
      </WrapperTypography>
    </div>
  );
};

export default Loading;
