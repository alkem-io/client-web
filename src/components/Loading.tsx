import React, { FC } from 'react';

export const Loading: FC = () => {
  return <div>Loading...</div>;
};
export default Loading;
