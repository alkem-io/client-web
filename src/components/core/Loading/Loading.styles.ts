import { createStyles } from '../../../hooks/useTheme';

export const useLoadingStyles = createStyles(theme => ({
  spinner: {
    color: theme.palette.primary,
  },
  text: {
    marginLeft: theme.shape.spacing(2),
  },
  container: {
    display: 'flex',
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
  },
}));
