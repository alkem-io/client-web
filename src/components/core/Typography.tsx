import clsx from 'clsx';
import React, { FC, useLayoutEffect, useMemo, useRef } from 'react';
import { Palette, Typography as TypographyContract } from '../../themes';
import { createStyles } from '../../hooks/useTheme';
import _clamp from 'clamp-js';
import { replaceAll } from '../../utils/replaceAll';

const useTypographyStyles = createStyles(theme => ({
  h1: {
    fontFamily: theme.typography.h1.font,
    fontSize: theme.typography.h1.size,
  },
  h2: {
    fontFamily: theme.typography.h2.font,
    fontSize: theme.typography.h2.size,
    textTransform: 'uppercase',
    marginBottom: theme.shape.spacing(1),
  },
  h3: {
    fontFamily: theme.typography.h3.font,
    fontSize: theme.typography.h3.size,
    marginBottom: theme.shape.spacing(1),
  },
  h4: {
    fontFamily: theme.typography.h4.font,
    fontSize: theme.typography.h4.size,
    marginBottom: theme.shape.spacing(1),
  },
  h5: {
    fontFamily: theme.typography.h5.font,
    fontSize: theme.typography.h5.size,
  },
  caption: {
    fontFamily: theme.typography.caption.font,
    fontSize: theme.typography.caption.size,
    textTransform: 'uppercase',
  },
  body: {
    fontFamily: theme.typography.body.font,
    fontSize: theme.typography.body.size,
  },
  button: {
    fontFamily: theme.typography.button.font,
    fontSize: theme.typography.button.size,
    textTransform: 'uppercase',
  },
  primary: {
    color: theme.palette.primary,
  },
  positive: {
    color: theme.palette.positive,
  },
  neutral: {
    color: theme.palette.neutral,
  },
  neutralMedium: {
    color: theme.palette.neutralMedium,
  },
  neutralLight: {
    color: theme.palette.neutralLight,
  },
  negative: {
    color: theme.palette.negative,
  },
  divider: {
    color: theme.palette.divider,
  },
  background: {
    color: theme.palette.background,
  },
  inherit: {
    color: 'inherit',
  },
}));

interface FontWeight {
  regular: number;
  medium: number;
  boldLight: number;
  bold: number;
}

const fontWeight: FontWeight = {
  regular: 400,
  medium: 500,
  boldLight: 700,
  bold: 900,
};

interface TypographyProps extends Record<string, unknown> {
  variant?: keyof TypographyContract;
  className?: string;
  color?: keyof Palette | 'inherit';
  weight?: keyof FontWeight;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  as?: React.ComponentType<any> | string;
  classes?: unknown;
  clamp?: string | number;
}

const Typography: FC<TypographyProps> = ({
  variant = 'body',
  color = 'neutral',
  weight = 'medium',
  as: Component = 'div',
  className,
  classes,
  children,
  clamp = undefined,
  ...rest
}) => {
  const styles = useTypographyStyles(classes);
  const ref = useRef();

  useLayoutEffect(() => {
    if (clamp && ref?.current) {
      _clamp(ref.current, { clamp });
    }
  });

  const removeSlashes = (string: string) => replaceAll('\\"', '"', replaceAll('\\n', '<br/>', string));

  const plainText = children as string;

  const text = useMemo(() => {
    if (children instanceof Array) {
      return children.map(c => {
        if (typeof c === 'string') return <span key={c} dangerouslySetInnerHTML={{ __html: removeSlashes(c) }} />;
        return c;
      });
    }
    if (typeof children === 'string') {
      return <span dangerouslySetInnerHTML={{ __html: removeSlashes(plainText) }} />;
    }
    return children;
  }, [children]);

  return (
    <Component
      {...rest}
      className={clsx(styles[variant], styles[color], className)}
      style={{ fontWeight: fontWeight[weight] }}
      ref={ref}
    >
      {text}
    </Component>
  );
};

export default Typography;
