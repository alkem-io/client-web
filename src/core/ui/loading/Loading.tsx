import { Box, CircularProgress } from '@mui/material';
import { Caption } from '../typography';

export const Loading = ({ text = 'Loading' }: { text?: string }) => {
  return (
    <Box
      sx={{
        display: 'flex',
        flexGrow: 1,
        alignItems: 'center',
        justifyContent: 'center',
        height: '100%',
        gap: 2,
      }}
    >
      <CircularProgress sx={{ color: 'primary.main' }} />
      <Caption textTransform="uppercase" fontWeight="medium" color="primary.main">
        {text}
      </Caption>
    </Box>
  );
};

export default Loading;
