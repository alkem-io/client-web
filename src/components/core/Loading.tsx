import clsx from 'clsx';
import React, { FC } from 'react';
import { Spinner } from 'react-bootstrap';
import { createStyles } from '../../hooks/useTheme';
import Typography from './Typography';

interface LoadingProps {
  text?: string;
}

const useStyles = createStyles(theme => ({
  spinner: {
    color: theme.palette.primary,
  },
  text: {
    marginLeft: theme.shape.spacing(2),
  },
  itemsContentCenter: {
    alignItems: 'center',
    justifyContent: 'center',
  },
}));

export const Loading: FC<LoadingProps> = ({ text = 'Loading' }) => {
  const styles = useStyles();

  return (
    <div className={clsx('d-flex', 'flex-grow-1', styles.itemsContentCenter)}>
      <Spinner animation="grow" className={styles.spinner} />
      <Typography variant="caption" color="primary" className={styles.text}>
        {text}
      </Typography>
    </div>
  );
};
export default Loading;
