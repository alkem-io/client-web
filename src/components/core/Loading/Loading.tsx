import React, { FC } from 'react';
import { Spinner } from 'react-bootstrap';
import Typography from '../Typography';
import { useLoadingStyles } from './Loading.styles';

interface LoadingProps {
  text?: string;
}

export const Loading: FC<LoadingProps> = ({ text = 'Loading' }) => {
  const styles = useLoadingStyles();

  return (
    <div className={styles.container}>
      <Spinner animation="grow" className={styles.spinner} />
      <Typography variant="caption" color="primary" className={styles.text}>
        {text}
      </Typography>
    </div>
  );
};
export default Loading;
