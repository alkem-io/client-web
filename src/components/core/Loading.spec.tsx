import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';

import Loading from './Loading';
import { Spinner } from 'react-bootstrap';
import Typography from './Typography';

test('render correctly Spinner component', () => {
  const SpinnerComponent = mount(<Spinner animation="grow" />);
  expect(SpinnerComponent).toMatchSnapshot();
});

test('render correctly Typography component', () => {
  const TypographyComponent = mount(<Typography variant="caption" color="primary" />);
  expect(TypographyComponent).toMatchSnapshot();
});

test('render correctly loading component', () => {
  const text = 'Please wait';
  const TextComponent = renderer.create(<Loading text={text} />).toJSON();
  expect(TextComponent).toMatchSnapshot();
});

test('check loading with message', () => {
  const message = 'Loading indicator';
  const LoadingComp = mount(<Loading text={message} />);
  expect(LoadingComp.text()).toEqual(message);
});

test('check loading without message', () => {
  const message = '';
  const LoadingComp = mount(<Loading text={message} />);
  expect(LoadingComp.text()).toEqual(message);
});
