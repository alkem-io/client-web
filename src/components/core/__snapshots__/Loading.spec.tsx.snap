// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component render correctly Spinner component 1`] = `
<DocumentFragment>
  <div
    class="spinner-grow"
  />
</DocumentFragment>
`;

exports[`Loading component render correctly Typography component 1`] = `
<DocumentFragment>
  <div
    class="caption-0-2-6 primary-0-2-9"
    style="font-weight: 500;"
  />
</DocumentFragment>
`;

exports[`Loading component render correctly loading component 1`] = `
<div
  className="d-flex flex-grow-1 itemsContentCenter-0-2-20"
>
  <div
    className="spinner-0-2-18 spinner-grow"
  />
  <div
    className="caption-0-2-26 primary-0-2-29 text-0-2-19"
    style={
      Object {
        "fontWeight": 500,
      }
    }
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "Please wait",
        }
      }
    />
  </div>
</div>
`;
